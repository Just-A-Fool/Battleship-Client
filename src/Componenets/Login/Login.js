import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import './Login.css';

export default class Login extends Component {

  render() {
    return (
      <form className='loginform'>
        <div>
          <label htmlFor='login-username-input'>Username</label>
          <input
            id='login-username-input'
            name='username'
            required
          />
        </div>

        <div>
          <label htmlFor='login-password-input'>Password</label>
          <input
            id='login-password-input'
            name='password'
            required
          />
        </div>

        <button type='submit'>Login</button>
        <button>
          <Link to='/guest'>
            Guest
          </Link>
        </button> <br />

        <Link to='/signup'>Need to create an account?</Link>
      </form>

      // <footer>
      // Copyright Â© since 2020
      // </footer>
    );
  };
  
};


