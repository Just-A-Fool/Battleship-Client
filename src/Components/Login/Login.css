.btnLink a {
  color: red;
}

.btnLink {
  padding-top: 2rem;
}