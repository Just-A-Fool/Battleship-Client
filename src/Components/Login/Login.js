import React, { Component } from 'react';
import { Link, withRouter } from 'react-router-dom';
import AuthApiService from '../../Services/auth-api-service';
import { Input, Required, Label } from '../Form/Form';
import Button from '../Button/Button';
import TokenService from '../../Services/token-service';
import Banner from '../Banner/Banner';
import './Login.css';

class Login extends Component {
  static defaultProps = {
    onLoginSuccess: () => { }
  }

  state = { error: null }

  resetError = () => {
    this.setState({
      error:null
    })
  }

  firstInput = React.createRef()

  handleSubmit = ev => {
    ev.preventDefault();
    const { username, password } = ev.target;
    
    let user = username.value.split(' ').join('');

    AuthApiService.postLogin({
      username: user,
      password: password.value,
    })
      .then(res => {
        username.value = '';
        password.value = '';
        TokenService.saveAuthToken(res.authToken);
      
        this.props.history.push('/dashboard');
      })
      .catch(res => {
        this.setState({ error: res.error })
      });
  }

  componentDidMount() {
    document.title = 'Login Page'
    this.firstInput.current.focus()
  }

  render() {
    let errorMessage = this.state.error ? <p className='errorMessage'>{this.state.error}</p>: null;

    return (
      <div className='login'>
        <Banner />
        <h1>Login</h1>
        
        <form className='loginform' onSubmit={event => this.handleSubmit(event)} onChange={() => this.resetError()}>
          <div>
            <Label htmlFor='login-username-input'>Username<Required /></Label>
            <Input
              ref={this.firstInput}
              id='login-username-input'
              name='username'
              required
              aria-required
            />
          </div>

          <div>
            <Label htmlFor='login-password-input'>Password<Required /></Label>
            <Input
              id='login-password-input'
              name='password'
              type='password'
              required
              aria-required
            />
          </div>
          {errorMessage}
          <div className='loginbtn'>
            <Button type='submit'>Login</Button>
            {' '}
            
            <Button onClick={() => {
              this.props.history.push('/guest')
            }}> Guest
            </Button> <br />
          </div>

          <div className='btnLink'>
            <Link to='/signup'>Need to create an account?</Link>
          </div>
        </form>
      </div>
    );
  };
};

export default withRouter(Login);


