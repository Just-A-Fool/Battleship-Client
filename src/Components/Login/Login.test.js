import React from 'react';
import ReactDOM from 'react-dom';
import Login from './Login';
import { BrowserRouter } from 'react-router-dom';

describe('Login Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render
      (<BrowserRouter> 
        <Login /> 
      </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

});


