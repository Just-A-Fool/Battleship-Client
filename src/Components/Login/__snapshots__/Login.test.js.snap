// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component Displays the 'Login' component when rendered 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withRouter(Login)>
      <Route>
        <Login
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          location={
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            }
          }
          match={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
          onLoginSuccess={[Function]}
        >
          <div
            className="login"
          >
            <Banner>
              <div
                className="banner"
              >
                <h1>
                  BATTLESHIP
                </h1>
              </div>
            </Banner>
            <h1>
              Login
            </h1>
            <form
              className="loginform"
              onChange={[Function]}
              onSubmit={[Function]}
            >
              <div>
                <Label
                  htmlFor="login-username-input"
                >
                  <label
                    className="Label"
                    htmlFor="login-username-input"
                  >
                    Username
                    <Required>
                      <span
                        className="Required"
                      >
                        *
                      </span>
                    </Required>
                  </label>
                </Label>
                <ForwardRef
                  aria-required={true}
                  id="login-username-input"
                  name="username"
                  required={true}
                >
                  <input
                    aria-required={true}
                    className="Input"
                    id="login-username-input"
                    name="username"
                    required={true}
                    type="text"
                  />
                </ForwardRef>
              </div>
              <div>
                <Label
                  htmlFor="login-password-input"
                >
                  <label
                    className="Label"
                    htmlFor="login-password-input"
                  >
                    Password
                    <Required>
                      <span
                        className="Required"
                      >
                        *
                      </span>
                    </Required>
                  </label>
                </Label>
                <ForwardRef
                  aria-required={true}
                  id="login-password-input"
                  name="password"
                  required={true}
                  type="password"
                >
                  <input
                    aria-required={true}
                    className="Input"
                    id="login-password-input"
                    name="password"
                    required={true}
                    type="password"
                  />
                </ForwardRef>
              </div>
              <div
                className="loginbtn"
              >
                <ForwardRef
                  type="submit"
                >
                  <button
                    className="Button"
                    type="submit"
                  >
                    Login
                  </button>
                </ForwardRef>
              </div>
              <div
                className="btnLink"
              >
                <Link
                  replace={false}
                  to="/signup"
                >
                  <a
                    href="/signup"
                    onClick={[Function]}
                  >
                    Need to create an account?
                  </a>
                </Link>
              </div>
            </form>
          </div>
        </Login>
      </Route>
    </withRouter(Login)>
  </Router>
</BrowserRouter>
`;
